src/planner/plan/append_extend.cpp

```diff
@@ -122,6 +128,71 @@ void Planner::appendNonRecursiveExtend(const std::shared_ptr<NodeExpression>& bo
     }
 }
 
+void Planner::appendRecursiveExtendAsGDS(const std::shared_ptr<NodeExpression>& boundNode,
+    const std::shared_ptr<NodeExpression>& nbrNode, const std::shared_ptr<RelExpression>& rel,
+    ExtendDirection direction, LogicalPlan& plan) {
+    // GDS pipeline
+    auto recursiveInfo = rel->getRecursiveInfo();
+    auto graphEntry =
+        graph::GraphEntry(recursiveInfo->node->getTableIDs(), recursiveInfo->rel->getTableIDs());
+    auto functionSet = VarLenJoinsFunction::getFunctionSet();
+    KU_ASSERT(functionSet.size() == 1);
+    auto gdsFunction = functionSet[0]->constPtrCast<GDSFunction>()->copy();
+    auto bindData = std::make_unique<RJBindData>(boundNode, nbrNode, recursiveInfo->lowerBound,
+        recursiveInfo->upperBound, direction);
+    bindData->setExtendFromSource(*boundNode == *rel->getSrcNode());
+    gdsFunction.gds->setBindData(std::move(bindData));
+    expression_vector resultColumns;
+    resultColumns.push_back(boundNode->getInternalID());
+    resultColumns.push_back(nbrNode->getInternalID());
+    if (direction == common::ExtendDirection::BOTH) {
+        resultColumns.push_back(recursiveInfo->pathEdgeDirectionsExpr);
+    }
+    resultColumns.push_back(recursiveInfo->lengthExpression);
+    resultColumns.push_back(recursiveInfo->pathNodeIDsExpr);
+    resultColumns.push_back(recursiveInfo->pathEdgeIDsExpr);
+    auto gdsInfo =
+        BoundGDSCallInfo(gdsFunction.copy(), graphEntry.copy(), std::move(resultColumns));
```

Comment:
Not directly related to this PR, but I found it a bit weird that planner is fixing the `resultColumns` here for the GDS call. By also looking at `Binder::bindInQueryCall`, cannot the function (table/gds func) figure out its own outputExprs?

Reply:
This is fixed.